File: src/_1ProgrammingBasicsWithJavaMay2023/_2ConditionalStatements/_2Exercise/_04ToyShop.java
package _1ProgrammingBasicsWithJavaMay2023._2ConditionalStatements._2Exercise;

import java.util.Scanner;

public class _04ToyShop {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double tripPrice = Double.parseDouble(scanner.nextLine());
        int puzzleCount = Integer.parseInt(scanner.nextLine());
        int dollsCount = Integer.parseInt(scanner.nextLine());
        int teddiesCount = Integer.parseInt(scanner.nextLine());
        int minionsCount = Integer.parseInt(scanner.nextLine());
        int trucksCount = Integer.parseInt(scanner.nextLine());
        double puzzlePrice = 2.60;
        double dollsPrice = 3;
        double teddiesPrice = 4.10;
        double minionsPrice = 8.20;
        double truckPrice = 2;
        double totalSum = (puzzlePrice * puzzleCount) + (dollsPrice * dollsCount) + (teddiesPrice * teddiesCount) + (minionsPrice * minionsCount) + (truckPrice * trucksCount);
        int totalCountToys = puzzleCount + dollsCount + teddiesCount + minionsCount + trucksCount;
        if (totalCountToys >= 50){
            totalSum = totalSum * 0.75;
        }
        totalSum = totalSum * 0.90;
        double diff = Math.abs(totalSum - tripPrice);
        if (totalSum >= tripPrice){
            System.out.printf("Yes! %.2f lv left.", diff);
        } else {
            System.out.printf("Not enough money! %.2f lv needed.",diff);
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_2ConditionalStatements/_2Exercise/_04ToyShop.js
function toyShop(priceForTrip, numberOfPuzzles, numberOfDolls, numberOfBears, numberOfMinions, numberOfTrucks) {
    const PUZZLE_PRICE = 2.60;
    const DOLL_PRICE = 3.00;
    const BEAR_PRICE = 4.10;
    const MINION_PRICE = 8.20;
    const TRUCK_PRICE = 2.00;

    const DISCOUNT_PERCENT_FOR_BUYING_MORE_THAN_50_TOYS = 0.25;
    const PERCENT_FOR_RENT = 0.10;

    const numberOfToys = numberOfPuzzles + numberOfDolls + numberOfBears + numberOfMinions + numberOfTrucks;

    const puzzlePrice = numberOfPuzzles * PUZZLE_PRICE;
    const dollPrice = numberOfDolls * DOLL_PRICE;
    const bearPrice = numberOfBears * BEAR_PRICE;
    const minionPrice = numberOfMinions * MINION_PRICE;
    const truckPrice = numberOfTrucks * TRUCK_PRICE;

    let totalPrice = puzzlePrice + dollPrice + bearPrice + minionPrice + truckPrice;

    if (numberOfToys >= 50) {
        totalPrice -= totalPrice * DISCOUNT_PERCENT_FOR_BUYING_MORE_THAN_50_TOYS;
    }

    totalPrice -= totalPrice * PERCENT_FOR_RENT;

    const diff = Math.abs(totalPrice - priceForTrip);
    if (totalPrice >= priceForTrip) {
        console.log(`Yes! ${diff.toFixed(2)} lv left.`);
    } else {
        console.log(`Not enough money! ${diff.toFixed(2)} lv needed.`);
    }

}